### PR TITLE
fix: User cannot send files from share extension more than 10 mb

### PR DESCRIPTION
**Pull request by agisilaos**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

To be added

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios-data-model/Tests/Source/Model/Messages/ZMAssetClientMessageTests.swift [hunk 1]
```diff
@@ -249,7 +249,7 @@ extension ZMAssetClientMessageTests {
 
         let otrKey = Data.randomEncryptionKey()
         let encryptedData = data.zmEncryptPrefixingPlainTextIV(key: otrKey)
-        let sha256 = encryptedData.zmSHA256Digest()
+        let sha256 = encryptedData!.zmSHA256Digest()
```
Comment (netbe) [inline wire-ios-data-model/Tests/Source/Model/Messages/ZMAssetClientMessageTests.swift]: maybe better to safely unwrap the value and do a XCTFail() in order not to crash the tests

```suggestion
        guard let encryptedData = encryptedData else {
          XCTFail("expected encrypted data") 
          return 
        }
        let sha256 = encryptedData.zmSHA256Digest()
```

File: wire-ios-request-strategy/Sources/Request Strategies/Assets/AssetV3DownloadRequestStrategyTests.swift [hunk 2]
```diff
@@ -492,7 +494,9 @@ extension AssetV3DownloadRequestStrategyTests {
         // GIVEN
         let plainTextData = Data.secureRandomData(length: 500)
         let key = Data.randomEncryptionKey()
-        let encryptedData = plainTextData.zmEncryptPrefixingPlainTextIV(key: key)
+        guard let encryptedData = plainTextData.zmEncryptPrefixingPlainTextIV(key: key) else {
+            return
```
Comment (netbe) [inline wire-ios-request-strategy/Sources/Request Strategies/Assets/AssetV3DownloadRequestStrategyTests.swift:500]: add fail test XCTFail...

File: wire-ios-request-strategy/Sources/Request Strategies/Assets/AssetV3DownloadRequestStrategyTests.swift [hunk 3]
```diff
@@ -596,14 +602,16 @@ extension AssetV3DownloadRequestStrategyTests {
...
+                                   + "<rect width=\"100\" height=\"100\"/>"
+                                   + "</svg>").data(using: .utf8) else {
+            XCTFail("Unable to convert SVG to Data")
+            return
         }
 
         let key = Data.randomEncryptionKey()
-        let encryptedData = plainTextData.zmEncryptPrefixingPlainTextIV(key: key)
+        guard let encryptedData = plainTextData.zmEncryptPrefixingPlainTextIV(key: key) else {
+            return
```
Comment (netbe) [inline wire-ios-request-strategy/Sources/Request Strategies/Assets/AssetV3DownloadRequestStrategyTests.swift:617]: same

File: wire-ios-request-strategy/Sources/Request Strategies/Assets/AssetV3DownloadRequestStrategyTests.swift [hunk 4]
```diff
@@ -531,7 +535,9 @@ extension AssetV3DownloadRequestStrategyTests {
     func testThatItRecategorizeMessageAfterDownloadingAssetContent() {
         let plainTextData = self.verySmallJPEGData()
         let key = Data.randomEncryptionKey()
-        let encryptedData = plainTextData.zmEncryptPrefixingPlainTextIV(key: key)
+        guard let encryptedData = plainTextData.zmEncryptPrefixingPlainTextIV(key: key) else {
+            return
```
Comment (netbe) [inline wire-ios-request-strategy/Sources/Request Strategies/Assets/AssetV3DownloadRequestStrategyTests.swift:542]: same

File: wire-ios-request-strategy/Sources/Request Strategies/Assets/AssetV3PreviewDownloadRequestStrategyTests.swift [hunk 5]
```diff
@@ -320,7 +320,9 @@ class AssetV3PreviewDownloadRequestStrategyTests: MessagingTestBase {
         // GIVEN
         let plainTextData = Data.secureRandomData(length: 500)
         let key = Data.randomEncryptionKey()
-        let encryptedData = plainTextData.zmEncryptPrefixingPlainTextIV(key: key)
+        guard let encryptedData = plainTextData.zmEncryptPrefixingPlainTextIV(key: key) else {
+            return
```
Comment (netbe) [inline wire-ios-request-strategy/Sources/Request Strategies/Assets/AssetV3PreviewDownloadRequestStrategyTests.swift:326]: same

File: wire-ios-utilities/Tests/Source/NSData+ZMSCryptoTests.swift [hunk 6]
```diff
@@ -75,7 +77,9 @@ extension NSData_ZMSCryptoTests {
 
         // when
         for _ in 0..<100 {
-            let data = sampleData.zmEncryptPrefixingPlainTextIV(key: self.sampleKey)
+            guard let data = sampleData.zmEncryptPrefixingPlainTextIV(key: self.sampleKey) else {
+                return
```
Comment (netbe) [inline wire-ios-utilities/Tests/Source/NSData+ZMSCryptoTests.swift:83]: XCTFail

**Review by netbe — COMMENTED** [review]
just some suggestion about unwrapping

File: wire-ios-utilities/Source/Public/Data+ZMSCrypto.swift [hunk 7]
```diff
@@ -78,8 +79,45 @@ public extension Data {
...
     }
 
-    func zmEncryptPrefixingPlainTextIV(key: Data) -> Data {
-        return (self as NSData).zmEncryptPrefixingPlainTextIV(withKey: key)
+    func zmEncryptPrefixingPlainTextIV(key: Data) -> Data? {
+
+        let keyLength = key.count
+        guard keyLength == kCCKeySizeAES256 else {
+            return nil
+        }
```
Comment (johnxnguyen) [inline wire-ios-utilities/Source/Public/Data+ZMSCrypto.swift]: Returning `nil` won't help the caller know what went wrong, should we consider making this a throwing function instead?

**Review by johnxnguyen — COMMENTED** [review]
I suggest to add a performance test to assert this change in behavior, see https://developer.apple.com/documentation/xctest/xctmemorymetric

File: wire-ios-data-model/Tests/Source/Model/Messages/ZMAssetClientMessageTests.swift (hunk 1 above)
Comment (agisilaos) [reply wire-ios-data-model/Tests/Source/Model/Messages/ZMAssetClientMessageTests.swift]: Done

File: wire-ios-request-strategy/Sources/Request Strategies/Assets/AssetV3DownloadRequestStrategyTests.swift (hunk 2 above)
Comment (agisilaos) [reply wire-ios-request-strategy/Sources/Request Strategies/Assets/AssetV3DownloadRequestStrategyTests.swift:500]: Done!

File: wire-ios-request-strategy/Sources/Request Strategies/Assets/AssetV3DownloadRequestStrategyTests.swift (hunk 3 above)
Comment (agisilaos) [reply wire-ios-request-strategy/Sources/Request Strategies/Assets/AssetV3DownloadRequestStrategyTests.swift:617]: Done!

File: wire-ios-request-strategy/Sources/Request Strategies/Assets/AssetV3DownloadRequestStrategyTests.swift (hunk 4 above)
Comment (agisilaos) [reply wire-ios-request-strategy/Sources/Request Strategies/Assets/AssetV3DownloadRequestStrategyTests.swift:542]: Done!

File: wire-ios-request-strategy/Sources/Request Strategies/Assets/AssetV3PreviewDownloadRequestStrategyTests.swift (hunk 5 above)
Comment (agisilaos) [reply wire-ios-request-strategy/Sources/Request Strategies/Assets/AssetV3PreviewDownloadRequestStrategyTests.swift:326]: Done!

File: wire-ios-utilities/Tests/Source/NSData+ZMSCryptoTests.swift (hunk 6 above)
Comment (agisilaos) [reply wire-ios-utilities/Tests/Source/NSData+ZMSCryptoTests.swift:83]: Done

File: wire-ios-utilities/Source/Public/Data+ZMSCrypto.swift (hunk 7 above)
Comment (agisilaos) [reply wire-ios-utilities/Source/Public/Data+ZMSCrypto.swift]: Fixed!

**Pull request closed without merging**